### ISSUE TITLE
no-useless-cast false positive

### ISSUE THREAD

**Issue by eranshabi** [SonarSource/SonarTS#834]
**I want to report a bug.**

**SonarTS version:** 1.9.0<br>
**Node.js version:** 10<br>
**TypeScript version:** 3.3.3333<br>
**TSLint version:** 5.13.1<br>

**Rule key:** no-useless-cast
**Reproducer** 
```
// minimal reproducer if relevant
  await page.waitFor(() => (document.querySelector('[data-hook="wix-media-image"]') as HTMLImageElement).complete); // False-positive here
```

**Logs**
```
ERROR: 8:17   no-useless-cast  Remove this unnecessary cast.
```

**Expected behavior**
cast is not useless because document.querySelector returns `Element` which has no `complete` attribute, so it needs to be cast to HTMLImageElement

**Comment by saberduck** [comment]
hello @eranshabi , 

I am not able to reproduce this issue with TypeScript 3.4.5 . This rule just uses information from TypeScript compiler, and consequently, it will depend on the type definitions for DOM. Can you try to upgrade your TypeScript version and see if the issue persists?
